csharp/device/Microsoft.Azure.Devices.Client/Message.cs

```diff
@@ -128,7 +158,11 @@ public string MessageId
         {
             get
             {
+#if NETMF
+                return (string)(this.GetSystemProperty(MessageSystemPropertyNames.MessageId) ?? string.Empty);
```

Comment:
I've replaced the code as you suggesting it's working and returning a null when the MessageId property doesn't exist.